environs/__init__.py

```diff
@@ -276,29 +276,71 @@ def read_env(
         file is found. If you do not wish to recurse up the tree, you may pass
         False as a second positional argument.
         """
-        # By default, start search from the same file this function is called
-        if path is None:
+        PATH_SEPARATOR_PATTERN = r"\\|\/"
+
+        if path and os.path.isdir(str(path)):
+            raise ValueError(
+                "The specified path '" + str(path) + "' is a directory. Ensure that you pass a filename."
+            )
+
+        original_path = path
+
+        # use .env as path if path is None or empty
+        if not path:
+            path = ".env"
+
+        # make path absolute if it is relative, using the directory of the calling script as base
+        if not os.path.isabs(path):
+
+            def merge_absolute_and_relative_path(absolute_path, relative_path):
```

Comment:
thought: It might make sense to disallow passing a path and `recurse=True`. That would obviate this function, right?

Reply:
That function was written due to some unit tests that were written before I started with the bugfix. Removing it might create headaches for some people, so I'd leave it in.